src/main/java/assignment/LaunchInterceptorConditions.java

```diff
@@ -41,23 +41,109 @@ public static boolean ConditionThree(double AREA1, int[][] POINTS) {
     }
 
     public static boolean ConditionFour(int QPTS, int QUADS, int[][] POINTS) {
-        // do something
-        return true;
+        if(QPTS < 2 || QPTS > POINTS.length){
+            return false;
+        }
+    
+        if(QUADS < 1 || QUADS > 3){
+            return false;
+        }
+
+        for(int i = 0; i < POINTS.length - QPTS -1; i++){
+            boolean[] FoundQuads = {false, false, false, false};
```

Comment:
I would like us to follow Java naming conventions. Methods and variables are in lowerCamelCase, classess are UpperCamelCase, constants are in UPPER_SNAKE_CASE. so it would be nice to rename FoundQuads into foundQuads.